sw/device/tests/rstmgr_alert_info_test.c

```diff
@@ -0,0 +1,685 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/math.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_alert_handler.h"
+#include "sw/device/lib/dif/dif_aon_timer.h"
+#include "sw/device/lib/dif/dif_i2c.h"
+#include "sw/device/lib/dif/dif_otp_ctrl.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/dif/dif_rv_timer.h"
+#include "sw/device/lib/dif/dif_spi_host.h"
+#include "sw/device/lib/dif/dif_uart.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/alert_handler_testutils.h"
+#include "sw/device/lib/testing/aon_timer_testutils.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/FreeRTOSConfig.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+#include "sw/device/lib/testing/test_framework/test_status.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+/*
+  RSTMGR ALERT_INFO Test
+
+  This test runs some alert / dump scenario and check integrity of
+  ALERT_INFO coming from alert_hander to rstmgr.
+
+  From dif_rstmgr_reset_info_t, there are 9 different resets.
+  This test covers 3 of those (kDifRstmgrResetInfoSw,
+  kDifRstmgrResetInfoWatchdog, kDifRstmgrResetInfoEscalation).
+  kDifRstmgrResetInfoNdm will be added as a separate test after ndm request test
+  is available. Test goes for 3 rounds and each round it creates different reset
+  scenario.
+
+  Round 1: Single Class profile
+  - Trigger alert from i2c0..2 by calling alert_force.
+  - This will trigger kDifAlertHandlerIrqClassa from alert_handler.
```

Comment:
It probably should be, however since this enum value is autogenerated from [these](https://cs.opensource.google/opentitan/opentitan/+/master:hw/top_earlgrey/ip_autogen/alert_handler/data/alert_handler.hjson;drc=55b5b7e7cd586512d7f50f2eee77b1fefafd310a;l=403), which are autogenerated from [this](https://cs.opensource.google/opentitan/opentitan/+/master:hw/ip_templates/alert_handler/data/alert_handler.hjson.tpl;drc=0805fc07bfca86a16fedaa4973ccebcb2badf267;l=300). We could fix this, but probably best done in a separate PR.